### ISSUE TITLE
defaultValue not handled in InheritInverseConfiguration

### ISSUE THREAD

**Issue by sgrekov** [mapstruct/mapstruct#1531]

**Comment by sjaakd** [comment]
behaviour is intentional. Only: `source`, `target` and if both are specified `ignore` is mapped. The reason is twofold:
1. A conversion to a type from and to `String` must exist (as of 1.3 primitive types and native types are treated as such) must be present.
2. There's no guarantee that `source` and `target` are of the same type.
3. There's no guarantee that the user wants the same for both `source` and `target`.

Having said that.. 3. could possibly be overcome in the same way as is specified above: specifying `source` and `target` in combination with `defaultValue`

**Comment by sgrekov** [comment]
> 2. There's no guarantee that source and target are of the same type.
Well, defaultValue() already assumes that 1. and 2. is true

By 
>  specifying source and target in combination with defaultValue
you mean drop using @InheritInverseConfiguration and do twoway mapping manually?

**Comment by sjaakd** [comment]
> Well, defaultValue() already assumes that 1. and 2. is true

Why? The problem is from a MapStruct perspective we cannot assume that a user always want a `defaultValue` to be applied when you inherit inverse mappings. The same goes for other fields you can specify in `@Mapping` besides `source` and `target` (like `qualifiedBy`, `dateFormat`, etc.) with the exception of `ignore` (explained below)


> you mean drop using @inheritinverseconfiguration and do twoway mapping manually?
 
No. That is not what I mean. However, you always have to option to inherit all inverse mappings but override individual mappings. So like this:
```java
@InheritInverseConfiguration // inherits all inverse mappings
@Mapping( source = "x", target = "y", defaultValue = "blah" ) // with exception of the mapping for target "y"
B mapTo( A dto );
```

That should solve your case. Let me know.

Forget about my remark: What I meant  was  a similar approach that we have to `ignore=true`. Only when you specify `source`, `target` and `ignore` the `ignore` is automatically applied to the `source`. So why is `ignore` treated differently than other fields in `@Mapping?:  `ignore` does apply to the `target`. Therefore: `source` only has meaning in the reverse mapping so in (`@InheritInverseConfiguration`). So, my mistake: This scenario is not comparable to what you want.

**Comment by sgrekov** [comment]
> you always have to option to inherit all inverse mappings but override individual mappings

Ah! This is what I want, thanks! It will be nice if you provide this example somewhere in docs more clearly, cause I didn't even assumed I can do it that way.

**Comment by sjaakd** [comment]
There's a line in `@InheritConfiguration` in the doc: *A method can use @InheritConfiguration and override or amend the configuration by additionally applying @Mapping, @BeanMapping, etc*. Its not there for `@InheritInverseConfiguration` and there's no example. 

So: you are right.. we could improve the documentation. If you have time: could you provide a PR with a doc update? I'll leave this one open as documentation PR

**Comment by sjaakd** [comment]
closed by PR #1535
